addon/state-for.js

```diff
@@ -27,11 +27,16 @@ export default function stateFor(stateName, propertyName) {
     return weakMaps[stateName];
   }
 
-  assert('The second argument must be a string', typeof(propertyName) === 'string');
+  assert('The second argument must be a string', typeof propertyName  === 'string');
 
   return computed(propertyName, function() {
     let propertyValue = this.get(propertyName);
 
+    assert(`The property '${propertyName}' must be an object, function, or array`,
```

Comment:
so following existing ember patterns, `null` `undefined` keys, should just return `null` `undefined`